applications/vanilla/controllers/api/DraftsApiController.php

```diff
@@ -0,0 +1,394 @@
+        $rows = $this->draftModel->getWhere($where, '', 'asc', $limit, $offset)->resultArray();
+
+        foreach ($rows as &$row) {
+            $this->prepareRow($row);
+        }
+
+        $result = $out->validate($rows);
+        return $result;
+    }
+
+    /**
+     * Update a draft.
+     *
+     * @param int $id The unique ID of the draft.
+     * @param array $body The request body.
+     * @return array
+     */
+    public function patch($id, array $body) {
+        $this->permission('Garden.SignIn.Allow');
+
+        $in = $this->draftPostSchema('in')->setDescription('Update a draft.');
+        $out = $this->schema(['draftID', 'parentRecordID', 'attributes'], 'out')->add($this->fullSchema());
+
+        $row = $this->draftByID($id);
+        if ($row['InsertUserID'] !== $this->getSession()->UserID) {
+            $this->permission('Garden.Moderation.Manage');
+        }
+
+        $body = $in->validate($body, true);
+        $recordType = !empty('DiscussionID') ? 'comment' : 'discussion';
+        $draftData = $this->translateRequest($body, $recordType);
+        $draftData['DraftID'] = $id;
+        $this->draftModel->save($draftData);
+        $this->validateModel($this->draftModel);
+
+        $updatedRow = $this->draftByID($id);
+        $this->prepareRow($updatedRow);
+
+        $result = $out->validate($updatedRow);
+        return $result;
+    }
+
+    /**
+     * Create a draft.
+     *
+     * @param array $body The request body.
+     * @return array
+     */
+    public function post(array $body) {
+        $this->permission('Garden.SignIn.Allow');
+
+        $in = $this->draftPostSchema('in')->setDescription('Create a draft.');
+        $out = $this->schema($this->fullSchema(), 'out');
+
+        $body = $in->validate($body);
+        $draftData = $this->translateRequest($body);
+        $draftID = $this->draftModel->save($draftData);
+        $this->validateModel($this->draftModel);
+
+        $row = $this->draftByID($draftID);
+        $this->prepareRow($row);
+
+        $result = $out->validate($row);
+        return $result;
+    }
+
+    /**
+     * Prepare data for output.
+     *
+     * @param array $row
+     */
+    public function prepareRow(array &$row) {
+        $row = $this->translateRow($row);
+    }
+
+    /**
+     * Translate a request to this endpoint into a format compatible for saving with DraftModel.
+     *
+     * @param array $body
+     * @param string $recordType
+     * @return array
+     */
+    private function translateRequest(array $body, $recordType = null) {
+        // If the record type is not explicitly defined by the parameters, try to extract it from $body.
+        if ($recordType === null && array_key_exists('recordType', $body)) {
+            $recordType = $body['recordType'];
+        }
+
+        if (array_key_exists('attributes', $body)) {
+            $columns = ['announce', 'body', 'categoryID', 'closed', 'format', 'name', 'sink', 'tags'];
```

Comment:
I think that these should be fetch from the relevant record's schema.

Reply:
These are actually just coming off the drafts table. They wouldn't change, based on the resource type. It's pulling whitelisted "drafts" columns and extracting them from submitted `attributes` property values, before saving.

Reply:
Most of them are bound to the discussion record.
The draft table should not impact the API endpoint.

Reply:
body, format, name are the only fields that SHOULD be common among all draft types.

Reply:
This is for translating a request into a format that can be passed to `DraftModel::save`. It has to follow the drafts table format. When things are refactored and the drafts table is structured like the API schema, then we can ditch this translation layer.

Reply:
Sorry I got confused.. I was referring to https://github.com/vanilla/vanilla/pull/6018/files/67bb66c03b353c2cebd8295d69ced5e210558003#r143755372

Reply:
The description and the validation should be done using the recordType schema ^